galsim/wcs.py

```diff
@@ -233,12 +253,47 @@ def posToWorld(self, image_pos, color=None, **kwargs):
                                 than returning a CelestialCoord. [default: None]
         @param projection       If project_center != None, the kind of projection to use.  See
                                 CelestialCoord.project for the valid options. [default: 'gnomonic']
+
+        @returns world_pos
         """
         if color is None: color = self._color
         if not isinstance(image_pos, Position):
             raise TypeError("image_pos must be a PositionD or PositionI argument")
         return self._posToWorld(image_pos, color=color, **kwargs)
 
+    def xyToWorld(self, x, y, units=None, color=None):
+        """Convert x,y from image coordinates to world coordinates.
+
+        This is equivalent to `wcs.toWorld(x,y)`.
+
+        It is also equivalent to `wcs.posToWorld(galsim.PositionD(x,y))` when x and y are scalars;
+        however, this routine allows x and y to be numpy arrays, in which case, the calculation
+        will be vectorized, which is often much faster than using the pos interface.
+
+        @param x            The x value(s) in image coordinates
+        @param y            The y value(s) in image coordinates
+        @param units        (Only valid for CelestialWCS, in which case it is required)
+                            The units to use for the returned ra, dec values.
+        @param color        For color-dependent WCS's, the color term to use. [default: None]
+
+        @returns ra, dec
+        """
+        from .angle import AngleUnit
+        if color is None: color = self._color
+        if self.isCelestial():
+            if units is None:
+                raise TypeError("units is required for CelestialWCS types")
+            elif isinstance(units, str):
+                units = AngleUnit.from_name(units)
+            elif not isinstance(units, AngleUnit):
+                raise GalSimValueError("units must be either an AngleUnit or a string", units,
+                                       AngleUnit.valid_names)
+            return self._xyToWorld(x, y, units, color)
+        else:
+            if units is not None:
+                raise TypeError("units is not valid for EuclideanWCS types")
```

Comment:
"Arbitrary" WCS is almost not really a thing.  Euclidean and Celestial WCS's are really pretty different beasts.  
I suspect it's more likely for this to help people identify when they thought they should be getting ra, dec back (with units) and accidentally passed a local WCS where that doesn't make sense, than that this would cause hardships of them wanting to use the same function call for both euclidean and celestial wcs types.